src/contentscript/swarm-library/local-storage.ts

```diff
@@ -0,0 +1,73 @@
+import { nanoid } from 'nanoid'
+import { ILocalStorageMessage } from '../../utils/message/local-storage'
+import { InpageReqMessageFormat, InterceptorResMessageFormat } from '../../utils/message/message-handler'
+import { MessengerInpage } from './messenger.inpage'
+
+export class LocalStorage extends MessengerInpage implements ILocalStorageMessage {
+  /** The real Bee API address that shouldn't be called directly by dApps */
+  public setItem(keyName: string, keyValue: unknown): Promise<void> {
+    const message: InpageReqMessageFormat<Parameters<ILocalStorageMessage['setItem']>> = {
+      key: 'setItem',
+      eventId: nanoid(),
+      sender: 'inpage',
+      target: 'content',
+      sessionId: window.swarmSessionId,
+      payload: [keyName, keyValue],
+    }
+
+    return new Promise<void>((resolve, reject) => {
+      const handler = (response: MessageEvent<InterceptorResMessageFormat<string>>) => {
+        if (!this.validMessage(response, message.eventId)) return
```

Comment:
Shouldn't this reject instead of return?

Reply:
no, because there are more handlers of `onMessage`